transaction/core/src/mint/validation/config.rs

```diff
@@ -194,14 +248,49 @@ mod tests {
             mint_limit: 15,
         };
 
-        assert_eq!(
-            validate_configs(token_id, &[mint_config1]),
-            Err(Error::InvalidSignerSet)
-        );
-        assert_eq!(
-            validate_configs(token_id, &[mint_config2]),
-            Err(Error::InvalidSignerSet)
-        );
+        for block_version in BlockVersion::iterator() {
+            if !block_version.mint_transactions_are_supported() {
+                continue;
+            }
+
+            assert_eq!(
+                validate_configs(token_id, &[mint_config1.clone()], block_version),
+                Err(Error::InvalidSignerSet)
+            );
+            assert_eq!(
+                validate_configs(token_id, &[mint_config2.clone()], block_version),
+                Err(Error::InvalidSignerSet)
+            );
+        }
+    }
+
+    #[test]
+    fn validate_configs_rejects_nested_multisigners_on_unsupported_block_versions() {
+        let mut rng = get_seeded_rng();
+        let signer_1 = Ed25519Pair::from_random(&mut rng);
+
+        let mint_config = MintConfig {
+            token_id: 123,
+            signer_set: SignerSet::new_with_multi(
+                vec![],
+                vec![SignerSet::new(vec![signer_1.public_key()], 1)],
+                1,
+            ),
+            mint_limit: 10,
+        };
+
+        for block_version in BlockVersion::iterator() {
+            if !block_version.mint_transactions_are_supported()
+                || block_version.nested_multisigs_are_supported()
+            {
+                continue;
+            }
+
+            assert_eq!(
+                validate_configs(123.into(), &[mint_config.clone()], block_version),
+                Err(Error::InvalidSignerSet)
+            );
+        }
```

Comment:
Unfortunately it's not yet guaranteed that this is the only version, since if this doesn't make it into the next release there might be two versions that did not support nesting. Hence why I prefer to rely on the specific properties of each version.